sanity-io/structured-content-2022
# Program redesign

# Program redesign

## Intent

Implements the redesigned `programSection` per updated Figma sketches.

Solves [this Shortcut Story](https://app.shortcut.com/sanity-io/story/16054/program-page-front-end).

## Details

This PR adds a semi-hardcoded `program?venue=<VENUE_NAME>` entry to _[[...slug]].tsx_' `getStaticPaths` for each Venue referenced by a Program. This currently correlates with the fact that the `programsSection` is rendered at _/program_, but if the program is ever moved to another location, this pre-generation will break; hence "semi-hardcoded".

## Testing this PR

1. Go to [/program](https://structured-content-2022-web-git-program-redesign.sanity.build/program) for [this PR's Preview Deploy URL](https://structured-content-2022-web-git-program-redesign.sanity.build)
2. Verify that the rendered elements correspond with the [Figma sketches](https://www.figma.com/file/XjAvJKsLpxYJSc8QDWK81j/Sanity-SCC-Website-(Shared)?node-id=7958%3A22278) for all specified viewports
3. Verify that you can select other venues from the "VenueNav" component, and that switching between Programs for the various Venues happens quickly (indicating that the various Program pages are pre-generated)
